### ISSUE TITLE
bug: Transactions disappear when the app is killed

### ISSUE THREAD

**Issue by fracali** [rarfell/dimeApp#35]
### Prerequisites

- [X] I have searched for [existing issues](https://github.com/rarfell/dimeApp/issues) that already report this problem, without success.

### Dime Version

2.1.3 (12)

### Current Behavior

Sometimes when I add transactions they appear in the list but when i kill the app completely and reopen it they disappear. I have the week insights widget on my home screen and I noticed that the transactions that disappear are never added to the widget. 

### Expected Behavior

The transactions should remain when you reopen the app

### Steps to Reproduce

I haven't found a way to reproduce this bug, I just noticed that every time a new transaction doesn't get added to the widget it disappears from the list the next time I open the app.

**Comment by Harishwarrior** [comment]
I am also facing this same issue. I have added a transaction and switched to another app. The OS killed the dime app which is in background and when I switched back to it, last transactions are gone!

**Comment by Harishwarrior** [comment]
I have also noticed when opening the app at the start of the day (After long time in background/killed). The app takes some times to load the entries. A loading indicator would be useful here to prevent inserting data when fetching the latest records. @rarfell

**Comment by FuNian788** [comment]
Same problem

**Comment by changsheng1239** [comment]
yes, can confirm this does happen, very likely due to icloud sync issue

**Comment by krtkush** [comment]
Same issue here. Dime loses some transactions randomly, but mostly around the end of the month or when I re-start my phone.

**Comment by fx-ha** [comment]
Did an iOS update and after the phone restarted I lost two weeks of transactions. App is not really usable with this issue

**Comment by sobri909** [comment]
For people having this issue, try turning off iCloud Sync in Dime's Settings. The most likely cause is bugs in iOS's CloudKit sync layer, which is used for the database syncing.

**Comment by makeavish** [comment]
Same issue 
Lost data a month of data😥

**Comment by tyelf22** [comment]
I keep running into the same issue.
